shared/components/data/themes-list-fetcher/index.jsx

```diff
@@ -1,44 +1,45 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	omit = require( 'lodash/object/omit' ),
-	once = require( 'lodash/function/once' ),
-	isEqual = require( 'lodash/lang/isEqual' );
+import React from 'react';
+import pick from 'lodash/object/pick';
+import omit from 'lodash/object/omit';
+import once from 'lodash/function/once';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
 
 /**
  * Internal dependencies
  */
-var ThemesStore = require( 'lib/themes/stores/themes' ),
-	ThemesListStore = require( 'lib/themes/stores/themes-list' ),
-	Actions = require( 'lib/themes/flux-actions' ),
-	Constants = require( 'lib/themes/constants' );
-
-function queryThemes( props ) {
-	Actions.query( {
-		search: props.search,
-		tier: props.tier,
-		page: 0,
-		perPage: Constants.PER_PAGE
-	} );
-
-	Actions.fetchNextPage( props.site );
-}
-
-function getThemesInList() {
-	return ThemesListStore.getThemesList().map( ThemesStore.getById );
-}
-
-function getThemesState() {
+import Constants from 'lib/themes/constants';
+import * as allActions from 'lib/themes/actions';
```

Comment:
I think I remember now, it was likely because of this:

<img width="603" alt="screen shot 2015-12-08 at 02 14 56" src="https://cloud.githubusercontent.com/assets/150562/11646040/9d6853bc-9d51-11e5-9acb-f2385ef2e4cb.png">

Between adding an `eslint-disable-line no-shadow` comment and doing `{ query: boundQuery, fetchNextPage: boundFetchNextPage } = this.props`, I don't know which is uglier. But reverting to calling `this.props.query` seems like a shame. In a way, the `allActions / pick` approach _kind_ of made sense. Thoughts? :)